docs/versioned_docs/version-zenith/developer-guide/typescript/817191-custom-types.mdx

```diff
@@ -0,0 +1,34 @@
+---
+slug: /developer-guide/typescript/817191/custom-types
+displayed_sidebar: 'zenith'
+toc_max_heading_level: 2
+title: 'Custom Types'
+---
+
+import PartialExperimentalDocs from '../../partials/_experimental.mdx';
+
+# Custom Types
+
+<PartialExperimentalDocs />
+
+TypeScript has multiple ways to support complex data types. However, the Dagger TypeScript SDK only supports it through fields (decorated with `@field()`) inside classes because `type` and `interface` keywords do not allows decorators.
+
+Here is an example of a function named `daggerOrganization` that returns a custom `GitHubOrganization` type, itself containing a collection of `GitHubAccount` types:
```

Comment:
Users shouldn't prefix their custom types with the name of the module since the API already does that. I don't think we should encourage users to do it by seeing it in examples. In the Python version for this doc I removed the prefix for this reason. Otherwise you'll see in the API these types:
- GitHub
- GitHubGitHubOrganization
- GitHubGitHubAccount